client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetIndexTemplatesV2ResponseTests.java

```diff
@@ -0,0 +1,141 @@
+ * under the License.
+ */
+
+package org.elasticsearch.client.indices;
+
+import org.elasticsearch.cluster.metadata.AliasMetadata;
+import org.elasticsearch.cluster.metadata.IndexTemplateV2;
+import org.elasticsearch.cluster.metadata.Template;
+import org.elasticsearch.common.compress.CompressedXContent;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.test.AbstractXContentTestCase.xContentTester;
+
+public class GetIndexTemplatesV2ResponseTests extends ESTestCase {
+
+    public void testFromXContent() throws Exception {
+        xContentTester(
+            this::createParser,
+            GetIndexTemplatesV2ResponseTests::createTestInstance,
+            GetIndexTemplatesV2ResponseTests::toXContent,
+            GetIndexTemplatesV2Response::fromXContent)
+            .supportsUnknownFields(true)
+            .randomFieldsExcludeFilter(a -> true)
+            .test();
+    }
+
+    private static GetIndexTemplatesV2Response createTestInstance() {
+        Map<String, IndexTemplateV2> templates = new HashMap<>();
+        if (randomBoolean()) {
+            int count = randomInt(10);
+            for (int i = 0; i < count; i++) {
+                templates.put(randomAlphaOfLength(10), randomTemplate());
+            }
+        }
+        return new GetIndexTemplatesV2Response(templates);
+    }
+
+    private static void toXContent(GetIndexTemplatesV2Response response, XContentBuilder builder) throws IOException {
+        builder.startObject();
+        builder.startArray("index_templates");
+        for (Map.Entry<String, IndexTemplateV2> e : response.getIndexTemplates().entrySet()) {
+            builder.startObject();
+            builder.field("name", e.getKey());
+            builder.field("index_template");
+            e.getValue().toXContent(builder, null);
+            builder.endObject();
+        }
+        builder.endArray();
+        builder.endObject();
+    }
+
+    private static IndexTemplateV2 randomTemplate() {
+        Settings settings = null;
+        CompressedXContent mappings = null;
+        Map<String, AliasMetadata> aliases = null;
+        List<String> patterns = Arrays.asList(generateRandomStringArray(10, 10, false, false));
+        List<String> composedOf = null;
+        if (randomBoolean()) {
+            settings = randomSettings();
+        }
+        if (randomBoolean()) {
+            mappings = randomMappings();
+        }
+        if (randomBoolean()) {
+            aliases = randomAliases();
+        }
+        if (randomBoolean()) {
+            composedOf = Arrays.asList(generateRandomStringArray(10, 10, false, false));
+        }
+        Template template = new Template(settings, mappings, aliases);
```

Comment:
I think these methods (the randomization) can be reused from the component template tests you already added to the HLRC?

Reply:
Yep, that's a good idea, I've changed it